x-pack/metricbeat/mb/lightmodules_test.go

```diff
@@ -0,0 +1,296 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+// +build !integration
+
+package mb
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/logp"
+	"github.com/elastic/beats/metricbeat/mb"
+)
+
+// TestLightModulesAsModuleSource checks that registry correctly lists
+// metricsets when used with light modules
+func TestLightModulesAsModuleSource(t *testing.T) {
+	logp.TestingSetup()
+
+	type testMetricSet struct {
+		name       string
+		module     string
+		isDefault  bool
+		hostParser mb.HostParser
+	}
+
+	cases := []struct {
+		title                     string
+		registered                []testMetricSet
+		expectedMetricSets        map[string][]string
+		expectedDefaultMetricSets map[string][]string
+	}{
```

Comment:
nit:

```cases := map[string]struct {
   ...
} {
  "no registered modules": {
  },
  "same module registered (mixed module case)": {
  },
  ...
}
```

This pattern makes definitions a little nicer and order is randomized on every run.

Reply:
I'll change this :+1: